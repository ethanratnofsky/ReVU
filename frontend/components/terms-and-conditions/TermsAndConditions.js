import { useState } from "react";
import { Image, Pressable, SafeAreaView, ScrollView, Text } from "react-native";
import Checkbox from "expo-checkbox";

import Button from "../button/Button";

import termsAndConditionsStyles from "./TermsAndConditionsStyles";

import { TERMS_AND_CONDITIONS } from "../../constants";

const TermsAndConditions = ({ navigation, route }) => {
    const [isAccepted, setIsAccepted] = useState(false);

    const firstTime = route.params?.firstTime || false;
    const id = route.params.id;
    const email = route.params.email;

    const handleButtonPress = () => {
        navigation.navigate("Home", { id: id, email });
    };

    return (
        <SafeAreaView style={termsAndConditionsStyles.container}>
            <Image
                style={termsAndConditionsStyles.logo}
                source={require("../../assets/images/revu-logo.png")}
            />
            <Text style={termsAndConditionsStyles.heading}>Terms & Conditions</Text>
            <Text style={termsAndConditionsStyles.subheading}>Updated November 11, 2022</Text>
            <ScrollView style={termsAndConditionsStyles.scrollContainer}>
                <Text style={termsAndConditionsStyles.text}>{TERMS_AND_CONDITIONS}</Text>
            </ScrollView>
            {firstTime && (
                <Pressable
                    style={termsAndConditionsStyles.acceptInputContainer}
                    onPress={() => setIsAccepted((prev) => !prev)}
                >
                    <Checkbox
                        style={termsAndConditionsStyles.checkbox}
                        disabled={false}
                        value={isAccepted}
                        onValueChange={setIsAccepted}
                    />
                    <Text style={termsAndConditionsStyles.agreeText}>
                        I have read and agree to ReVU's terms and conditions.
                    </Text>
                </Pressable>
            )}
            <Button
                disabled={firstTime && !isAccepted}
                fontSize={14}
                style={termsAndConditionsStyles.button}
                onPress={handleButtonPress}
                text={firstTime ? "Got it!" : "Ah, yes, I remember!"}
            />
        </SafeAreaView>
    );
};

export default TermsAndConditions;
