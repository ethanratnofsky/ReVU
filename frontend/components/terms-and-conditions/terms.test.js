import React from 'react';
const renderer = require('react-test-renderer');
import TermsAndConditions from './TermsAndConditions';

describe('<TermsAndConditions />', () => {
  it('has 5 children', () => {
    const tree = renderer.create(<TermsAndConditions />).toJSON();
    expect(tree.children.length).toBe(5);
  });
});

it('renders correctly', () => {
    const tree = renderer.create(<TermsAndConditions />).toJSON();
    expect(tree).toMatchSnapshot();
  });