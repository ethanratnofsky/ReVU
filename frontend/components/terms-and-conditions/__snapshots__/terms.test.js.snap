// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#1C1C1C",
      "height": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 50,
        "marginBottom": 20,
        "marginTop": 10,
        "width": 172,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontFamily": "LibreCaslonText_400Regular",
        "fontSize": 18,
        "marginBottom": 10,
      }
    }
  >
    Terms & Conditions
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 14,
        "marginBottom": 20,
        "opacity": 0.5,
      }
    }
  >
    Updated November 11, 2022
  </Text>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "width": "85%",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
          }
        }
      >
        By using this application, you agree to posting appropriate content with proper language or risk having your comments/complaints removed.
      </Text>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 45,
        "opacity": 1,
        "width": 165,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        Object {
          "colors": Array [
            4294897334,
            4290024008,
          ],
          "endPoint": Array [
            1,
            0,
          ],
          "locations": undefined,
          "startPoint": Array [
            0,
            1,
          ],
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "padding": 3,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 8,
            "flexDirection": "row",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#CFAE70",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          Ah, yes, I remember!
        </Text>
      </View>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</RCTSafeAreaView>
`;
